### PR TITLE
Modify popup content to be based on data layer 

### PR DESCRIPTION
**Pull request by kailamjeter**
## Description

Addresses: [display-water-system-data-on-the-map](https://app.shortcut.com/blueconduit/story/4784/display-water-system-data-on-the-map?cf_workflow=500000067&ct_workflow=all&vc_group_by=day)

Modify popup content to get data from data layer rather than have it hard coded.

This also contains changes to:
-  switch the state watcher to a watcher on state.currentDataLayer
- unify style and functions between data layers

### New

-  PopupInfo
- data_layer_util

### Changed

- Popup content
- State watcher in MapView

### Removed

- N/A

## Testing and Reviewing

npm run serve

## screenshot

Screen Shot 2022-06-13 at 3 31 37 PM

### CODE REVIEW

File: client/src/components/MapView.vue [hunk 1]
```diff
@@ -51,35 +52,58 @@ export default defineComponent({
...
   },
   methods: {
+    /**
+     * Sets the visibility of the layer with the given styleLayerId.
+     */
+    setDataLayerVisibility(styleLayerId: string, visible: boolean): void {
+      if (this.map == null) return;
+
+      const visibility = visible ? 'visible' : 'none';
+      this.map.setLayoutProperty(styleLayerId, 'visibility', visibility);
```
Comment (beekley) [inline client/src/components/MapView.vue:62]: nit: If I understand typescript correctly, I think this method (and similar ones below) could be a cute one-liner:

```
this.map?.setLayoutProperty(styleLayerId, 'visibility', visible ? 'visible' : 'none');
```

(but concise !== readable, so feel free to ignore this)

File: client/src/data_layer_configs/lead_and_copper_violations_by_county_config.ts [hunk 2]
```diff
@@ -1,48 +1,61 @@
...
+    visibility: 'none',
   },
 };
 
-const LEGEND_INFO: LegendInfo = {
-  title: 'Number of violations',
-  bucketMap: new Map(Object.entries(LEGEND_COLOR_MAP)),
+const popupInfo: PopupInfo = {
+  title: 'County',
+  subtitle: '320 estimated lead service lines',
```
Comment (beekley) [inline client/src/data_layer_configs/lead_and_copper_violations_by_county_config.ts:55]: is `320` significant here? or a placeholder?

**Review by beekley — APPROVED** [review]

File: client/src/components/MapView.vue (hunk 1 above)
Comment (kailamjeter) [reply client/src/components/MapView.vue:62]: Done.

File: client/src/data_layer_configs/lead_and_copper_violations_by_county_config.ts (hunk 2 above)
Comment (kailamjeter) [reply client/src/data_layer_configs/lead_and_copper_violations_by_county_config.ts:55]: Placeholder. Left a TODO to replace.

**Pull request merged**